cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/ResourceDistributionGoal.java

```diff
@@ -476,31 +476,47 @@ private boolean rebalanceByMovingLoadIn(Broker broker,
     while (!candidateBrokerPQ.isEmpty() && (actionType == INTER_BROKER_REPLICA_MOVEMENT
                                             || (actionType == LEADERSHIP_MOVEMENT && broker.leaderReplicas().size() != broker.replicas().size()))) {
       Broker cb = candidateBrokerPQ.poll();
-      SortedSet<Replica> candidateReplicasToReceive = cb.trackedSortedReplicas(replicaSortName).sortedReplicas(true);
-
-      for (Iterator<Replica> iterator = candidateReplicasToReceive.iterator(); iterator.hasNext(); ) {
-        Replica replica = iterator.next();
-        Broker b = maybeApplyBalancingAction(clusterModel, replica, Collections.singletonList(broker), actionType,
-                                             optimizedGoals, optimizationOptions);
-
-        // Only need to check status if the action is taken. This will also handle the case that the source broker
-        // has nothing to move in. In that case we will never reenqueue that source broker.
-        if (b != null) {
-          if (isLoadAboveBalanceLowerLimit(broker)) {
-            clusterModel.untrackSortedReplicas(replicaSortName);
-            return false;
-          }
-          // Remove the replica from its source broker if it was a replica movement.
-          if (actionType == INTER_BROKER_REPLICA_MOVEMENT) {
-            iterator.remove();
```

Comment:
Where do we remove source broker's replica in this PR? Or, maybe we don't need to remove the replica in the beginning?